File: scripts/all_items_from_a_collection/all_items_from_a_collection.py
import requests, math, json

ocApiUrl = 'https://oc-index.library.ubc.ca'
apiKey = '' # your api key here
collection = 'darwin'
perPage = 25
offset = 0

# Query the API for the collection item count
collectionUrl = ocApiUrl + '/collections/' + collection + '?api_key=' + apiKey
apiResponse = requests.get(collectionUrl).json()
itemCount = float(apiResponse['data']['items'])

# Figure out how many pages there are
pages = int(math.ceil(itemCount / float(perPage)))

# Loop through collection item pages to get all items
itemIds = []
for x in range(0, pages):
    collectionItemsUrl = ocApiUrl + '/collections/' + collection
    collectionItemsUrl += '/items?limit=' + str(perPage) + '&offset=' + str(offset) + '&api_key=' + apiKey
    offset += 25
    # Get list of 25 items
    apiResponse = requests.get(collectionItemsUrl).json()
    collectionItems = apiResponse['data']
    # Add each item id to the itemIds list
    for collectionItem in collectionItems:
        itemIds.append(collectionItem['_id'])

# Store all the items so we can print them out later
items = []
for itemId in itemIds:
    itemUrl = ocApiUrl + '/collections/' + collection + '/items/' + itemId
    apiResponse = requests.get(itemUrl).json()
    item = apiResponse['data']
    items.append(item)

print(json.dumps(items))

File: scripts/all_items_from_a_collection/all_items_from_a_collection.go
// Author: Yves Beaudoin
package main

import (
	"encoding/json"
	"log"
	"fmt"
	"github.com/christophwitzko/go-curl"
	"strings"
)

const (
    // Replace this public key with your API Key
	_apiKey = ""
    // Define the collection to get all items from
	_collection = "darwin"
)
// Set up the data types for unmarshalling the JSON search results
type _searchResult4Collection struct {
	DATA []struct {
		ID string `json:"_id"`
	} `json:"data"`
}
type _searchResult4ItemData struct {
	DATA interface{} `json:"data"`
}

func main() {
	var (
		collectionItems _searchResult4Collection
		err error
		itemData _searchResult4ItemData
		itemBytes       []byte
		items           [][]byte
	)

	// Get all the item ids for the collection
	curlOutput := curlGet("https://oc-index.library.ubc.ca/collections/" + _collection +
	"/items" +
	"?api_key=" + _apiKey)

	// Store them in a slice we can use
	if err = json.Unmarshal(curlOutput, &collectionItems); err != nil {
		log.Fatalln("\a[json.Unmarshal] ", err)
	}

	// Loop through each item id and store the associated JSON data
	for itemIdx, item := range collectionItems.DATA {
		fmt.Println("Item ID =", item.ID, strings.Repeat("-", 80))
		curlOutput = curlGet("https://oc-index.library.ubc.ca/collections/" + _collection +
		"/items/" + item.ID +
		"?api_key=" + _apiKey)

		if err = json.Unmarshal(curlOutput, &itemData); err != nil {
			log.Fatalln("\a[json.Unmarshal] ", err)
		}
		if itemBytes, err = json.MarshalIndent(itemData, "", "  "); err != nil {
			log.Fatalln("\a[json.MarshalIndent] ", err)
		}
		items = append(items, itemBytes)
		fmt.Println(string(items[itemIdx]))
	}
}
func curlGet(URL string) []byte {
	err, curlOutput, response := curl.Bytes(URL, "method=", "GET", "disablecompression=", true)
	if err != nil {
		log.Fatalln("\a[curlGet] ", err)
	} else if response != nil && response.StatusCode != 200 {
		log.Fatalln("\a[curlGet] response.StatusCode = ", response.StatusCode)
	}
	return curlOutput
}
